tests/run_test.go

```diff
@@ -0,0 +1,217 @@
+		}
+	}
+}
+
+func TestHttpPort(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	port := "4000"
+	logsCh := pebbleRun(t, pebbleDir, "--http=:"+port)
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	if !isPortUsedByProcess(t, port, "pebble") {
+		t.Errorf("Pebble is not listening on port %s", port)
+	}
+}
+
+func TestVerbose(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose")
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
+		"svc1:",
+		"override: replace",
+		"startup: enabled",
+		"command: /bin/sh -c",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+}
+
+func TestArgs(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose",
+		"--args",
+		"svc1",
+		"-c",
+		fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName)),
+	)
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
+		"svc1:",
+		"override: replace",
+		"startup: enabled",
+		"command: /bin/sh",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+}
+
+func TestIdentities(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	identitiesYAML := `
+identities:
+    bob:
+        access: admin
+        local:
+            user-id: 42
+    alice:
+        access: read
+        local:
+            user-id: 2000
+`
+	identitiesFileName := "idents-add.yaml"
+	createIdentitiesFile(t, pebbleDir, identitiesFileName, identitiesYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--identities="+filepath.Join(pebbleDir, identitiesFileName))
+	expectedLogs := []string{
+		"Started daemon",
+		"POST /v1/services",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	expectedOutput := []string{"access: admin", "local:", "user-id: 42"}
+	runPebbleCmdAndCheckOutput(t, pebbleDir, expectedOutput, "identity", "bob")
```

Comment:
This is another case where it's quite unclear what the helper is doing. I think we should refactor or maybe not even use a helper. If we do stick with a helper, I think it's much more natural for the helper to run the command and return the output (just as a string), then have the test check that:

```go
output := runPebbleCommand(t, pebbleDir, "identity", "bob")
expected := `
access: admin
local:
  user-id: 42
`[1:]
if output != expected { /* error */ }
```